cmd/infra/powervs/create.go

```diff
@@ -1438,3 +1470,128 @@ func attachTag(gtag *globaltaggingv1.GlobalTaggingV1, infraID string, resourceId
 
 	return nil
 }
+
+// setupTransitGateway set up the transit gateway for the cluster infra
+func (infra *Infra) setupTransitGateway(ctx context.Context, options *CreateInfraOptions, gtag *globaltaggingv1.GlobalTaggingV1) error {
+	log(options.InfraID).Info("Setting up Transit Gateway ...")
+
+	tgapisv1, err := transitgatewayapisv1.NewTransitGatewayApisV1(&transitgatewayapisv1.TransitGatewayApisV1Options{
+		Authenticator: getIAMAuth(),
+		Version:       utilpointer.String(currentDate),
+	})
+
+	if err != nil {
+		return err
+	}
+	var transitGateway *transitgatewayapisv1.TransitGateway
+	if options.TransitGateway != "" {
+		log(options.InfraID).Info("Validating Transit Gateway", "name", options.TransitGateway)
+		transitGateway, err = validateTransitGatewayByName(ctx, tgapisv1, options.TransitGateway, true)
+		if err != nil {
+			return fmt.Errorf("error validating transit gateway by name %s, %w", options.TransitGateway, err)
+		}
+	} else {
+		transitGateway, err = infra.createTransitGateway(ctx, options, tgapisv1)
+		if err != nil {
+			return fmt.Errorf("error creating transit gateway: %w", err)
+		}
+	}
+
+	if transitGateway != nil {
+		infra.TransitGatewayID = *transitGateway.ID
+		infra.Stats.TransitGatewayState.Status = *transitGateway.Status
+	} else {
+		return fmt.Errorf("unable to setup transit gateway")
+	}
+
+	if err = attachTag(gtag, options.InfraID, transitGateway.Crn, fmt.Sprintf("%s-%s", infra.ID, transitGatewayNameSuffix)); err != nil {
+		log(options.InfraID).Error(err, "error attaching tags to transit gateway")
+	}
+
+	log(options.InfraID).Info("Transit Gateway Ready", "id", infra.TransitGatewayID)
+	return nil
+}
+
+// createTransitGateway creates transit gateway and connections
+func (infra *Infra) createTransitGateway(ctx context.Context, options *CreateInfraOptions, tgapisv1 *transitgatewayapisv1.TransitGatewayApisV1) (*transitgatewayapisv1.TransitGateway, error) {
+
+	transitGatewayName := fmt.Sprintf("%s-%s", options.InfraID, transitGatewayNameSuffix)
+
+	tg, err := validateTransitGatewayByName(ctx, tgapisv1, transitGatewayName, true)
+	if err != nil && err.Error() != transitGatewayNotFound(transitGatewayName).Error() {
+		return nil, fmt.Errorf("error validating existing transit gateway: %w", err)
+	}
+	if tg != nil && *tg.Status == "available" {
+		log(options.InfraID).Info("Using existing transit gateway ...")
+		return tg, nil
```

Comment:
if available the does it mean that we also have connections as well? what if we just have the transit gateway but not the connections for vpc and powervs?

Reply:
I am validating the vpc and powervs connections are attached to this tg and their status is ok in `validateTransitGatewayByName` func. Then only I would return the tg obj and nil. If tg is there and connections are not proper then I return `transitGatewayConnectionError`.